ext/curl/share.c

```diff
@@ -134,6 +136,112 @@ PHP_FUNCTION(curl_share_strerror)
 }
 /* }}} */
 
+/**
+ * Initialize a persistent curl share handle with the given share options, reusing an existing one if found.
+ *
+ * Throws an exception if the share options are invalid.
+ */
+PHP_FUNCTION(curl_persistent_share_init)
+{
+	zval *share_opts = NULL, *entry = NULL;
+	zend_ulong persistent_id = 0;
+
+	php_curlsh *sh;
+
+	CURLSHcode error;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_ARRAY_EX(share_opts, 0, 1)
+	ZEND_PARSE_PARAMETERS_END();
+
+	object_init_ex(return_value, curl_persistent_share_ce);
+	sh = Z_CURL_SHARE_P(return_value);
+
+	ZEND_HASH_FOREACH_VAL(Z_ARRVAL_P(share_opts), entry) {
+		ZVAL_DEREF(entry);
+
+		zend_ulong option = zval_get_long_ex(entry, true);
```

Comment:
Forgive me since I'm not super familiar with the various zval functions: it seems like `zval_try_get_long` is fallible, but what we want is to fail hard if the zval isn't a long, right? Do you mean I should throw an exception myself via `zend_throw_exception_ex`?

I assumed the `is_strict` part of `zval_get_long_ex` would mean it fails fast - though reading the comment on my previous PR, it seems like that might be wrong.

Reply:
No this parameter only adds a warning when using a "float" that has a fractional part.

This function will happily convert a string like `"not an int"` to `0`, same with arrays, it basically acts like an `(int)` cast.

And yes, you will need to manually throw an exception, probably using `zend_argument_type_error`

Reply:
I believe I've addressed this, thank you!